src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1628,6 +1628,15 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                 body = null;
             }
 
+            // Delegate constructors and invokes have no body
+            if (method.IsExtern ||
```

Comment:
>if (method.IsExtern || [](start = 12, length = 22)

It feels like this check should be performed earlier in the pipe line. The place at the beginning of MethodCompiler.CompileMethod, which intercepts abstract methods, looks like the good place for it. It is fine to do that in a separate PR, or open an issue to follow-up. If we are going to open an issue, then at the very least this check should be moved to the very beginning of the function, there is no goo reason to go through all the things we do above if we are going to return null at the end.

Reply:
I just moved the check to the suggested place (beginning of MethodCompiler.CompileMethod). I removed the IsExtern check because it would cause us to produce different diagnostics, and we were getting a lot of tests failures as a result. Since we're simply fixing the behavior with SynthesizedDelegateSymbols, I think it's better to keep the extern checks as they were previously.

Reply:
Sounds fine.

Reply:
>  I removed the IsExtern check because it would cause us to produce different diagnostics, and we were getting a lot of tests failures as a result.

Opened https://github.com/dotnet/roslyn/issues/25658 to follow up on this.

Reply:
Good. I needed to remove the IsExtern from the assert in BindConstructorInitializerIfAny. I put it back in if again.